src/main/java/me/jellysquid/mods/lithium/mixin/math/sine_lut/MixinMathHelper.java

```diff
@@ -1,12 +1,35 @@
 package me.jellysquid.mods.lithium.mixin.math.sine_lut;
 
 import me.jellysquid.mods.lithium.common.util.math.CompactSineLUT;
+import net.fabricmc.fabric.api.event.lifecycle.v1.ServerLifecycleEvents;
 import net.minecraft.util.math.MathHelper;
+
 import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Mutable;
 import org.spongepowered.asm.mixin.Overwrite;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
 
 @Mixin(MathHelper.class)
 public class MixinMathHelper {
+    @Shadow
+    @Mutable
+    public static float[] SINE_TABLE;
+
+    @Inject(method = "<clinit>", at = @At("RETURN"))
+    private static void onClassInit(CallbackInfo ci) {
+        new CompactSineLUT(); // Force class initialisation
+        MixinMathHelper.SINE_TABLE = null;
+    }
+
+    static {
+        ServerLifecycleEvents.SERVER_STARTING.register((server) -> {
```

Comment:
The above injection should be sufficient already, I don't think this part is needed

Reply:
Yeah I'm new to Minecraft modding so I added both methods because I was unsure which one would be preferred.